packages/quantic/force-app/main/default/lwc/quanticCategoryFacet/e2e/quanticCategoryFacet.e2e.ts

```diff
@@ -0,0 +1,194 @@
+        const firstSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.clickOnFacetValue(selectedIndex);
+
+        await facetSelectUaRequest;
+        const firstSearchResponse = await firstSearchResponsePromise;
+        const {analytics: analyticsForFacetSelect} =
+          baseFacet.extractDataFromResponse(firstSearchResponse);
+        expect(analyticsForFacetSelect.actionCause).toEqual('facetSelect');
+        if (useCase.value === useCaseEnum.search) {
+          expect(analyticsForFacetSelect.customData).toEqual(
+            expect.objectContaining(expectedFacetData)
+          );
+        }
+
+        const FacetClearAllUaRequest =
+          baseFacet.waitForFacetClearAllUaAnalytics({
+            facetId,
+            facetField: field,
+          });
+        const secondSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.allCategoriesButton.click();
+
+        await FacetClearAllUaRequest;
+        const secondSearchResponse = await secondSearchResponsePromise;
+        const {analytics: analyticsForFacetDeselect} =
+          baseFacet.extractDataFromResponse(secondSearchResponse);
+        expect(analyticsForFacetDeselect.actionCause).toEqual('facetClearAll');
+      });
+    });
+
+    test.describe('when expanding and collapsing facet values using Show More/Less buttons', () => {
+      test.use({
+        options: {
+          numberOfValues: 2,
+        },
+      });
+
+      test('should fetch additional facet values and send correct UA analytics', async ({
+        baseFacet,
+        facet,
+      }) => {
+        const {facetId, field} = facetData;
+        const uaRequest = baseFacet.waitForShowMoreFacetResultsUaAnalytics({
+          facetId,
+          facetField: field,
+        });
+        const firstSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.showMoreFacetValuesButton.click();
+
+        await uaRequest;
+        const firstSearchResponse = await firstSearchResponsePromise;
+        const {analytics: analyticsForShowMore} =
+          baseFacet.extractDataFromResponse(firstSearchResponse);
+        expect(analyticsForShowMore.actionCause).toEqual(
+          'showMoreFacetResults'
+        );
+
+        const showLessUaRequest =
+          baseFacet.waitForShowLessFacetResultsUaAnalytics({
+            facetId,
+            facetField: field,
+          });
+        const secondSearchResponsePromise = baseFacet.waitForSearchResponse();
+
+        await facet.showLessFacetValuesButton.click();
+
+        await showLessUaRequest;
+        const secondSearchResponse = await secondSearchResponsePromise;
+        const {analytics: analyticsForShowLess} =
+          baseFacet.extractDataFromResponse(secondSearchResponse);
+        expect(analyticsForShowLess.actionCause).toEqual(
+          'showLessFacetResults'
+        );
+      });
+    });
+
+    test.describe('when a custom caption is provided', () => {
```

Comment:
Is this a e2e test?

I can see the analytics part to make sure for example that the caption isn't sent as part of the analytics.
But testing that the breadcrumb displays the custom caption though?

Reply:
it's basically tested this way because custom captions are passed via slots and slots are kinda tricky to handle with unit tests, so went with the e2e test for all what custom captions to ensure simplicity.